lib/context.js

```diff
@@ -105,7 +105,14 @@ const proto = module.exports = {
     // to node-style callbacks.
     if (null == err) return;
 
-    if (!(err instanceof Error)) err = new Error(`non-error thrown: ${err}`);
+    if (!(err instanceof Error)) {
+      try {
+        if (Object.prototype.toString.call(err) === '[object Object]') {
+          err = JSON.stringify(err);
```

Comment:
err must convert to Error instance here, BTW can we use `new Error(util.format('non-error thrown: %j',  err)` here?

Reply:
What is the reason to convert it to an error if on the next line it would be converted to a string with `${err}`? And generally there is no sense in the conversion because stack trace is already lost - new Error would have a stack trace to Koa internals which is not helpful :( Although usage of `util.format` may be a good idea - I'll try to play with it to make a code nicer

Reply:
it will emit in `app` [here](https://github.com/koajs/koa/pull/1113/files/7a2889efccde3df6035390f7d1886c09bad21720#diff-d911385a221c0b7502eabe1fdac9d209R123), we must keep `err` to be an instance of Error so it won't break anyone's code.

Reply:
Hm. But it would already always become an error [two lines lower](https://github.com/koajs/koa/pull/1113/files/7a2889efccde3df6035390f7d1886c09bad21720#diff-d911385a221c0b7502eabe1fdac9d209R114)

Reply:
You where right about `util.format` - it really covers all this cases without `if`s and additional code! Consider a new version now